backend/TaskTree/Models/Task.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.ComponentModel.DataAnnotations;
+
+namespace TaskTree.Models
+{
+	[Index(nameof(Name), IsUnique = true)]
```

Comment:
Don't think there's a need for task names to be unique, they are still distinguishable from their hierarchy and unique IDs. I could envision there being multiple tasks that have duplicate subtasks